cloudwatch_logs_common/src/utils/log_file_manager.cpp

```diff
@@ -77,16 +69,19 @@ FileObject<LogCollection> LogFileManager::readBatch(
       log_data.push_front(input_event);
       data_tokens.push_back(new_data_token);
     }
-    else{
-      AWS_LOG_INFO(__func__, "Some logs were not batched since the time"
-        " difference was > 24 hours. Will try again in a separate batch./n"
-        "Logs read: %d, Logs batched: %d", batch_size, log_data.size()
-        );
-      break;
+    else if(latestTime - curTime > TWO_WEEK_IN_SEC){
+      //how to discard the old logs?
```

Comment:
See `file_manager.h` `FileManager::fileUploadCompleteStatus` and notice how each token is resolved. You could do that if the option to discard old logs is set.